app/src/main/java/com/github/dedis/student20_pop/ui/event/PollEventCreationFragment.java

```diff
@@ -0,0 +1,196 @@
+        pollType.setOnCheckedChangeListener(new RadioGroup.OnCheckedChangeListener() {
+            @Override
+            public void onCheckedChanged(RadioGroup group, int checkedId) {
+                pollTypeIsOneOfN = (checkedId == R.id.radio_poll_type_1);
+            }
+        });
+
+        //Choices list
+        choicesListView = view.findViewById(R.id.choices_list);
+        listViewAdapter = new ChoicesListViewAdapter(this.getContext(), buttonsTextWatcher);
+        choicesListView.setAdapter(listViewAdapter);
+        justifyListViewHeightBasedOnChildren(choicesListView);
+
+        ImageButton addChoiceButton;
+        addChoiceButton = view.findViewById(R.id.button_add);
+        addChoiceButton.setFocusable(false);
+        addChoiceButton.setOnClickListener(clicked -> {
+            listViewAdapter.addChoice();
+            justifyListViewHeightBasedOnChildren(choicesListView);
+        });
+
+
+        // formatting today's date
+        try {
+            today = DATE_FORMAT.parse(DATE_FORMAT.format(Calendar.getInstance().getTime()));
+        } catch (ParseException e) {
+            e.printStackTrace();
+            today = new Date();
+        }
+
+        // Schedule
+        scheduleButton = view.findViewById(R.id.schedule_button);
+        scheduleButton.setOnClickListener(clicked -> {
+            PoPApplication app = (PoPApplication) (getActivity().getApplication());
+            String question = questionEditText.getText().toString();
+            ArrayList<String> choicesList = getChoices(choicesListView);
+            Event pollEvent = new PollEvent(question,
+                                            choicesList,
+                                            pollTypeIsOneOfN,
+                                            startDate,
+                                            endDate,
+                                            startTime,
+                                            endTime,
+                                            app.getCurrentLao().getId(),
+                                            "");
+            eventCreatedListener.OnEventCreatedListener(pollEvent);
+
+            fragmentManager.popBackStackImmediate();
+        });
+
+        Button cancelButton = view.findViewById(R.id.cancel_button);
+        cancelButton.setOnClickListener(clicked -> {
+            fragmentManager.popBackStackImmediate();
+        });
+        return view;
+    }
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+        checkDates(requestCode, resultCode, data);
+    }
+
+    private boolean isScheduleButtonEnabled() {
+        String question = questionEditText.getText().toString().trim();
+        int numberOfChoices = getChoices(choicesListView).size();
+
+        return !question.isEmpty() && numberOfChoices >= 2;
+    }
+
+    //TODO: make it a set so can't have the same choice twice
```

Comment:
I don't see a problem in letting the organizer create a poll with two similar choices. I don't see how it would be used, but there is no reason to block it.

Reply:
true